src/ray/core_worker/core_worker.h

```diff
@@ -883,6 +883,13 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   void ExecuteTaskLocalMode(const TaskSpecification &task_spec,
                             const ActorID &actor_id = ActorID::Nil());
 
+  /// KillActor API for a local mode.
+  Status KillActorLocalMode(const ActorID &actor_id);
+
+  /// Get a handle to a named actor for local mode.
+  std::pair<const ActorHandle *, Status> GetNamedActorHandleLocalMode(
+      const std::string &name);
```

Comment:
It's a bit ugly to have these in the main core worker just for local mode - any good to way to separate it out into another class?

Reply:
I can technically make a class like LocalModeCoreWorker, but I thought it would be just a meaningless class because 
1. there are already lots of local mode code that are hard to be separated out from core worker. 
2. And LocalModeCoreWorker should be able to access some private members of core_worker. We can use callback hack, but I think it'll be adding unnecessary complexity for just a tiny feature. 

And already there are methods like `ExecuteTaskLocalMode` in the core worker. 

To fundamentally solve this issue, I think we should refactor core worker a lot to have correct hierarchy and abstraction. Something like 

`core_worker_interface` # API specs
`core_worker` # regular
`core_worker_local_mode` # local mode
`core_worker_receiver` # code around executeTask
`core_worker_controller` # business logic
`core_worker_shared_states` # accessor to shared states among threads

Reply:
Maybe good workaround here is to make a class like LocalModeNamedRegistry and move all methods into this class. If you want this approach, I can do it!